File: Tutorials and Problems/P2a/Solutions a/LunchList1b/LunchList1b/app/src/main/java/org/feup/apm/lunchlist1b/MainActivity.java
package org.feup.apm.lunchlist1b;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.RadioGroup;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    ActionBar bar = getSupportActionBar();
    if (bar != null) {
      bar.setIcon(R.drawable.rest_icon);
      bar.setDisplayShowHomeEnabled(true);
    }
    setContentView(R.layout.activity_main);
    findViewById(R.id.bt_save).setOnClickListener((View v)->onBtSaveClick());
  }

  void onBtSaveClick() {
    Restaurant r = new Restaurant();
    r.setName(((EditText)findViewById(R.id.ed_name)).getText().toString());
    r.setAddress(((EditText)findViewById(R.id.ed_address)).getText().toString());
    switch (((RadioGroup)findViewById(R.id.rg_types)).getCheckedRadioButtonId()) {
      case R.id.rb_take:
        r.setType("take");
        break;
      case R.id.rb_sit:
        r.setType("sit");
        break;
      case R.id.rb_delivery:
        r.setType("delivery");
        break;
    }
  }

}

File: Tutorials and Problems/3.1. The Lunch Restaurants App/LunchList1/LunchList1b/app/src/main/java/org/feup/apm/lunchlist1b/MainActivity.kt
package org.feup.apm.lunchlist1b

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    supportActionBar?.setIcon(R.drawable.rest_icon)
    supportActionBar?.setDisplayShowHomeEnabled(true)
    setContentView(R.layout.activity_main)
    findViewById<Button>(R.id.bt_save).setOnClickListener { _ -> onBtSaveClick() }
  }

  private fun onBtSaveClick() {
    val r = Restaurant()
    r.name = findViewById<EditText>(R.id.ed_name).text.toString()
    r.address = findViewById<EditText>(R.id.ed_address).text.toString()
    r.type = when (findViewById<RadioGroup>(R.id.rg_types).checkedRadioButtonId) {
               R.id.rb_take -> "take"
               R.id.rb_sit -> "sit"
               R.id.rb_delivery -> "delivery"
               else -> ""
             }
  }
}

class Restaurant {
  var name = ""
  var address = ""
  var type = ""
}
